test/integration/test_team4_hform.cpp

```diff
@@ -182,3 +182,112 @@ TEST_CASE_METHOD(TestTEAM4HForm, "TestTEAM4HForm", "[CheckRun]")
   REQUIRE(peak_current_time > 0.01);
 }
+
+/// Test case for checking "Updates" work as expected following a mesh refinement.
+TEST_CASE_METHOD(TestTEAM4HForm, "TestTEAM4HFormMeshUpdates", "[CheckRun]")
+{
+  // Create Formulation
+  auto problem_builder = std::make_unique<hephaestus::HFormulation>(
+      "electric_resistivity", "electric_conductivity", "magnetic_permeability", "magnetic_field");
+  // Set Mesh
+  mfem::Mesh mesh((std::string(DATA_DIR) + std::string("./team4_symmetrized.g")).c_str(), 1, 1);
+  auto pmesh = std::make_shared<mfem::ParMesh>(MPI_COMM_WORLD, mesh);
+
+  problem_builder->SetMesh(pmesh);
+  problem_builder->AddFESpace("H1", "H1_3D_P1");
+  problem_builder->AddFESpace("HCurl", "ND_3D_P1");
+  problem_builder->AddFESpace("HDiv", "RT_3D_P0");
+  problem_builder->AddFESpace("Scalar_L2", "L2_3D_P0");
+  problem_builder->AddFESpace("Vector_L2", "L2_3D_P0", 3);
+  problem_builder->AddGridFunction("magnetic_field", "HCurl");
+
+  problem_builder->AddGridFunction("dmagnetic_potential_dt", "H1");
+  problem_builder->AddGridFunction("magnetic_flux_density", "HDiv");
+  problem_builder->RegisterMagneticFluxDensityAux("magnetic_flux_density");
+
+  problem_builder->AddGridFunction("current_density", "HDiv");
+  problem_builder->RegisterCurrentDensityAux("current_density");
+
+  problem_builder->AddGridFunction("electric_field", "HCurl");
+  problem_builder->RegisterElectricFieldAux("electric_field");
+
+  hephaestus::Coefficients coefficients = DefineCoefficients();
+  problem_builder->SetCoefficients(coefficients);
+
+  hephaestus::Sources sources = DefineSources();
+  problem_builder->SetSources(sources);
+
+  hephaestus::Outputs outputs = DefineOutputs();
+  problem_builder->SetOutputs(outputs);
+
+  problem_builder->AddBoundaryCondition("tangential_dhdt_bc",
+                                        std::make_shared<hephaestus::VectorDirichletBC>(
+                                            "dmagnetic_field_dt",
+                                            mfem::Array<int>({1, 2, 5, 6}),
+                                            coefficients._vectors.Get("surface_tangential_dHdt")));
+  problem_builder->AddBoundaryCondition(
+      "magnetic_potential_bc",
+      std::make_shared<hephaestus::ScalarDirichletBC>(
+          "dmagnetic_potential_dt",
+          mfem::Array<int>({1, 2}),
+          coefficients._scalars.Get("magnetic_potential_time_derivative")));
+
+  auto fluxmonitor = std::make_shared<hephaestus::FluxMonitorAux>("current_density", 3);
+  fluxmonitor->SetPriority(2);
+  problem_builder->AddPostprocessor("FluxMonitor", fluxmonitor);
+
+  hephaestus::InputParameters solver_options;
+  solver_options.SetParam("AbsTolerance", float(1.0e-20));
+  solver_options.SetParam("Tolerance", float(1.0e-20));
+  solver_options.SetParam("MaxIter", (unsigned int)500);
+  problem_builder->SetSolverOptions(solver_options);
+
+  problem_builder->FinalizeProblem();
+
+  auto problem = problem_builder->ReturnProblem();
```

Comment:
Could put the problem setup in a method and share it with the existing TEAM 4 test? That way we won't need to change it in two places if/when we make changes in future, and it should make these tests more readable